### PR TITLE
plant scraper complete

### PR DESCRIPTION
**Pull request by cchchoi1986**

### CODE REVIEW

**Comment by cchchoi1986** [comment]
@harrychenca @paulld @cassiemak

File: log/development.log [hunk 1]
```diff
@@ -17323,3 +17323,829 @@ Migrating to CreateRegionBirdLinks (20150215074708)
   [1m[36mSQL (0.3ms)[0m  [1mINSERT INTO "regions" ("created_at", "district", "name", "updated_at") VALUES (?, ?, ?, ?)[0m  [["created_at", "2015-02-15 08:26:18.975924"], ["district", "Hong Kong"], ["name", "Wan Chai"], ["updated_at", "2015-02-15 08:26:18.975924"]]
```
Comment (h8rry) [inline log/development.log]: What is development log still here?

**Comment by h8rry** [comment]
`tmp/cache/` files are also committed. We can't have that @cassiemak @cchchoi1986

File: lib/tasks/get_plants.rake [hunk 2]
```diff
@@ -4,6 +4,88 @@ namespace :get_plants do
...
+        chinese_name = ""
+        common_name = ""
+        plant_type = ""
+        flowering_period = ""
+
+        photos = get_photos(html_doc)
+
+        index = 1
+        html_doc.css("table.products_details_table tr").each do |row|
+          if row.text.gsub(/(Family Name)(.*)/,'\1')=="Family Name"
```
Comment (h8rry) [inline lib/tasks/get_plants.rake]: You can use 'case statements' to make this easier to read

File: lib/tasks/get_plants.rake [hunk 3]
```diff
@@ -4,6 +4,88 @@ namespace :get_plants do
...
+    subpages = (2360..13112).to_a
+    # subpages = (3339..3341).to_a
+    total = 0
+    subpages.each do |f|
+    html_doc = Nokogiri::HTML(open("http://www.herbarium.gov.hk/subpages.aspx?id=#{f}").read)
+      if html_doc.css("div.Text_overflow > table.products_details_table tr:nth-child(1) > td:nth-child(2)").text != " "
+        puts '========================================'
+        total = total + 1
+        # photos =[]
+        family_name = ""
```
Comment (h8rry) [inline lib/tasks/get_plants.rake]: You don't need to initialize variables for strings

File: lib/tasks/get_plants.rake [hunk 4]
```diff
@@ -4,6 +4,88 @@ namespace :get_plants do
     require 'open-uri'
     require 'nokogiri'
 
-    
+    subpages = (2360..13112).to_a
+    # subpages = (3339..3341).to_a
+    total = 0
+    subpages.each do |f|
+    html_doc = Nokogiri::HTML(open("http://www.herbarium.gov.hk/subpages.aspx?id=#{f}").read)
```
Comment (h8rry) [inline lib/tasks/get_plants.rake]: Indentation

File: lib/tasks/get_plants.rake [hunk 5]
```diff
@@ -4,6 +4,88 @@ namespace :get_plants do
...
+      puts "~total: #{total}"
+    end
+
+  end
+end
+
+def get_photos(html_doc)
+  photos =[]
+  imgScript = html_doc.css("script")[2].text
+  imgScript.gsub!(/(var imgArrSmall =\[)(.*)/,'')
```
Comment (h8rry) [inline lib/tasks/get_plants.rake:68]: So many gsubs?

File: lib/tasks/get_plants.rake [hunk 6]
```diff
@@ -4,6 +4,88 @@ namespace :get_plants do
     require 'open-uri'
     require 'nokogiri'
 
-    
+    subpages = (2360..13112).to_a
```
Comment (h8rry) [inline lib/tasks/get_plants.rake:7]: Interesting range of numbers...how did you find it?

**Comment by h8rry** [comment]
The rest looks okay

File: lib/tasks/get_plants.rake (hunk 5 above)
Comment (cchchoi1986) [inline lib/tasks/get_plants.rake:68]: yes even then there is still leftover "\" when shown in the hash

File: lib/tasks/get_plants.rake (hunk 6 above)
Comment (cchchoi1986) [inline lib/tasks/get_plants.rake:7]: ran it a few times

File: lib/tasks/get_plants.rake (hunk 4 above)
Comment (cchchoi1986) [inline lib/tasks/get_plants.rake]: fixed

File: lib/tasks/get_plants.rake (hunk 3 above)
Comment (cchchoi1986) [inline lib/tasks/get_plants.rake]: deleted

File: log/development.log (hunk 1 above)
Comment (cchchoi1986) [inline log/development.log]: how to get rid

File: lib/tasks/get_plants.rake (hunk 2 above)
Comment (cchchoi1986) [inline lib/tasks/get_plants.rake]: not sure what you mean here

**Comment by h8rry** [comment]
You need to remove tmp and development.log files

**Pull request merged**